src/views/BrowseInventory/BrowseInventory.js

```diff
@@ -6,18 +10,75 @@ import {
 } from '@folio/stripes-smart-components';
 import {
   FiltersPane,
+  ResetButton,
+  SingleSearchForm,
   useFiltersToogle,
+  useLocationFilters,
 } from '@folio/stripes-acq-components';
 
 import {
+  BrowseInventoryFilters,
   BrowseResultsPane,
   SearchModeNavigation,
 } from '../../components';
+import { browseInstanceIndexes } from '../../filterConfig';
+import {
+  useBrowseValidation,
+  useInventoryBrowse,
+} from '../../hooks';
 
 const BrowseInventory = () => {
+  const history = useHistory();
+  const location = useLocation();
+  const intl = useIntl();
   const [namespace] = useNamespace();
   const { isFiltersOpened, toggleFilters } = useFiltersToogle(`${namespace}/filters`);
 
+  const [
+    filters,
+    searchQuery,
+    applyFilters,
+    applySearch,
+    changeSearch,
+    resetFilters,
+    changeSearchIndex,
+    searchIndex,
+  ] = useLocationFilters(location, history, () => {});
+
+  const {
+    data,
+    isFetching,
+    isLoading,
+    pagination,
+    totalRecords,
+  } = useInventoryBrowse({
+    filters,
+    searchIndex,
+  });
+
+  const { validateDataQuery } = useBrowseValidation(searchIndex);
+
+  const searchableIndexesPlaceholder = intl.formatMessage({ id: 'ui-inventory.browse.searchableIndexesPlaceholder' });
+  const isResetButtonDisabled = !location.search && !searchQuery;
+
+  const formattedSearchableIndexes = useMemo(() => (
```

Comment:
I'm not sure that useMemo is required here. `browseInstanceIndexes` is static array with 3 elements, and memoization in this case will cost more than additional calculations

Reply:
Thanks! Looks like you're right, made updates.